test-runner/test_runner.py

```diff
@@ -187,30 +195,42 @@ def get_test_list(args: dict, tests_yaml: List[dict]):
         # returns the stdout of the test and the RETURNCODE
         try:  # Try for Runtime Failure Conditions
             log = test.container_run() if test.img else test.run()
-        except (DockerException, PerfException, YAMLError) as err:
+        except DockerException as err:
             logging.error(err)
             summary.append([idx + 1, test.name, "FAIL"])
+            json_summary.append(
+                {"Group": args.test_name, "Test": test.name, "Status": "FAIL"}
+            )
             ERROR = True
             continue
         except KeyboardInterrupt:
             summary.append([idx + 1, test.name, "FAIL"])
+            json_summary.append(
+                {"Group": args.test_name, "Test": test.name, "Status": "FAIL"}
+            )
             ERROR = True
             break
         summary.append([idx + 1, test.name, "PASS"])
+        json_summary.append(
+            {"Group": args.test_name, "Test": test.name, "Status": "PASS"}
+        )
+    json_summary_path = f"my-artifact-{unique_identifier}-test-runner.json"
+
+    with open(json_summary_path, "w") as file:
+        json.dump(json_summary, file, indent=4)
+
     # Switch logging context back to the initial state
     set_log_filename(logging.getLogger(), "test-runner", args.logs_path)
     # Remove remaining containers
-    test_images = [expandvars(test.img) for test in tests if test.img]
```

Comment:
This section and onward is reverting a previous PR that fixed related to removal of containers after running.

Reply:
Got it, I was previously using the deprecated `test_runner.py` I have reverted this changes to the new one.